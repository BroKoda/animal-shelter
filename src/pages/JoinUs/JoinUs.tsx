import React from 'react'
import { Link } from 'react-router-dom'
import BaseLayout from '../../layout/BaseLayout'

const JoinUs = (): JSX.Element => {
  return (
    <BaseLayout>
      <div className="container">
        <div className="row">
          <div className="col-12 mb-4">
            <h1>Jelentkezz önkéntesnek</h1>
          </div>
        </div>
        <div className="row mb-5">
          <div className="col-6 align-self-center">
            <p><strong>Cím: </strong></p>
            <p>
              Beogradska 3-5<br />
              24420 Magyarkanizsa<br />
              Szerbia
            </p>
            <p className='mt-3'><strong>Telefon: </strong></p>
            <p>
              +381 00 000 000
            </p>
            <p className='mt-3'><strong>Email: </strong></p>
            <p>
              mancsmagyarkanizsa@gmail.com
            </p>
            <p className='mt-3'><strong>Facebook: </strong></p>
            <p>
              <Link to={''}>
                Mancs Állatvédő Egyesület
              </Link>
            </p>
          </div>
          <div className="col-6">
            <form className="contact-form p-4">
              <h4 className="mb-3">Adataid:</h4>
              <div className="row">
                <div className="col-12">
                  <div className="form-control-container">
                    <div className="form-icon-container">
                      <i className="fa-solid fa-user"></i>
                    </div>
                    <input type="text" className="form-control" placeholder="Vezetéknév" id="contactLastName"/>
                  </div>
                </div>
                <div className="col-12">
                  <div className="form-control-container">
                    <div className="form-icon-container">
                      <i className="fa-solid fa-user"></i>
                    </div>
                    <input type="text" className="form-control" placeholder="Keresztnév" id="contactFirstName"/>
                  </div>
                </div>
                <div className="col-12">
                  <div className="form-control-container">
                    <div className="form-icon-container">
                      <i className="fa-solid fa-at"></i>
                    </div>
                    <input type="email" className="form-control" placeholder="Email" id="contactEmail"/>
                  </div>
                </div>
                <div className="col-12">
                  <div className="form-control-container">
                    <div className="form-icon-container">
                      <i className="fa-solid fa-phone"></i>
                    </div>
                    <input type="text" className="form-control" placeholder="Telefonszám" id="contactPhone"/>
                  </div>
                </div>
                <div className="col-12">
                  <div className="form-control-container">
                    <div className="form-icon-container">
                      <i className="fa-solid fa-message"></i>
                    </div>
                    <textarea className="form-control" placeholder="Üzenet" id="contactMessage"/>
                  </div>
                </div>
              </div>
              <div className="row justify-content-end">
                <div className="col-6 mt-3">
                  <button className="button call-to-action-button w-100">
                    Küldés
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
        <div className='row'>
          <div className="col-12">
            <h4 className='mb-4'>Támogass minked önkéntes adományoddal:</h4>
            <p><strong>Számla tulajdonos: </strong>Mancs Állatvédő Egyesület</p>
            <p className='mt-1'><strong>Bank neve: </strong>OTP Bank</p>
            <p className='mt-1'><strong>Belföldi számláról utalás: </strong>00000000 - 00000000 - 00000000</p>
            <p className='mt-1'><strong>Külfőldi számláról utalás: </strong>SRB62 00000000 - 00000000 - 00000000</p>
          </div>
        </div>
      </div>
    </BaseLayout>
  )
}

export default JoinUs