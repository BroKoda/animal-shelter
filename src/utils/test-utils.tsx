import React, { PropsWithChildren } from 'react'
import { render } from '@testing-library/react'
import type { RenderOptions } from '@testing-library/react'
import type { PreloadedState } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'
import { setupStore } from '../app/testStore'
import type { TestAppStore, TestRootReducer } from '../app/testStore'

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<TestRootReducer>
  store?: TestAppStore
}

export function renderWithProviders(
  ui: JSX.Element,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
): any {
  function Wrapper({ children }: PropsWithChildren<unknown>): JSX.Element {
    return <Provider store={store}>{children}</Provider>
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}