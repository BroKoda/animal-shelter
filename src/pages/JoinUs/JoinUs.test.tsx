import { renderWithProviders } from '../../utils/test-utils'
import { BrowserRouter } from 'react-router-dom'
import JoinUs from './JoinUs'

describe('Join us page test', () => {
  it('Should render join us page', async () => {
    const { container } = renderWithProviders(
      <BrowserRouter>
        <JoinUs />
      </BrowserRouter>
    )
    expect(container).toMatchSnapshot()
  })
})