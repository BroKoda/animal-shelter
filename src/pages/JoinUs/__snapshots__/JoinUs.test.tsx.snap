// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Join us page test Should render join us page 1`] = `
<div>
  <div
    class="container-fluid header-container"
  >
    <div
      class="row h-100"
    >
      <div
        class="col-12 align-self-center"
      >
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col-4 d-md-none align-self-center"
            >
              <i
                class="fa-solid fa-bars js-navigation-toggle"
              />
            </div>
            <div
              class="col-4 col-md-6 align-self-center text-center text-md-start"
            >
              <a
                href="/"
              >
                <img
                  alt="Mancs logo"
                  class="d-none d-md-block"
                  src="/mancs.png"
                />
                <img
                  alt="Mancs logo"
                  class="d-block d-md-none mx-auto"
                  src="/mancs_logo_only.png"
                />
              </a>
            </div>
            <div
              class="col-4 col-md-6 align-self-center text-end"
            >
              <a
                class="d-none d-md-inline-block"
                href="/regisztracio"
              >
                <button
                  class="button secondary-button me-2"
                >
                  Regisztráció
                </button>
              </a>
              <a
                href="/bejelentkezes"
              >
                <button
                  class="button call-to-action-button login-button"
                >
                  <span
                    class="d-none d-md-inline"
                  >
                    Bejelentkezés
                  </span>
                  <span
                    class="d-inline d-md-none"
                  >
                    <i
                      class="fa-solid fa-user"
                    />
                  </span>
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="container body-container mt-3 mb-5"
  >
    <div
      class="row"
    >
      <div
        class="col-12 col-md-4 col-lg-3 navigation-container d-none d-md-block pe-md-0 pe-xl-2"
      >
        <div
          class="container menu-container px-0"
        >
          <div
            class="row"
          >
            <div
              class="col-12"
            >
              <a
                href="/"
              >
                <div
                  class="menu-item"
                >
                  <div
                    class="menu-item-icon-container"
                  >
                    <i
                      class="fa-solid fa-house"
                    />
                  </div>
                  <span>
                    Főoldal
                  </span>
                </div>
              </a>
              <a
                href="/lakok"
              >
                <div
                  class="menu-item"
                >
                  <div
                    class="menu-item-icon-container"
                  >
                    <i
                      class="fa-solid fa-paw"
                    />
                  </div>
                  <span>
                    Lakóink
                  </span>
                </div>
              </a>
              <a
                href="/hirek"
              >
                <div
                  class="menu-item"
                >
                  <div
                    class="menu-item-icon-container"
                  >
                    <i
                      class="fa-solid fa-newspaper"
                    />
                  </div>
                  <span>
                    Hírek
                  </span>
                </div>
              </a>
              <a
                href="/csatlakozz"
              >
                <div
                  class="menu-item"
                >
                  <div
                    class="menu-item-icon-container"
                  >
                    <i
                      class="fa-solid fa-hand-holding-heart"
                    />
                  </div>
                  <span>
                    Csatlakozz!
                  </span>
                </div>
              </a>
              <a
                href="/kapcsolat"
              >
                <div
                  class="menu-item"
                >
                  <div
                    class="menu-item-icon-container"
                  >
                    <i
                      class="fa-solid fa-location-dot"
                    />
                  </div>
                  <span>
                    Kapcsolat
                  </span>
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-12 col-md-8 col-lg-9"
      >
        <div
          class="container body-container px-0"
        >
          <div
            class="row justify-content-center"
          >
            <div
              class="col-12 col-sm-10 col-md-12"
            >
              <div
                class="container"
              >
                <div
                  class="row"
                >
                  <div
                    class="col-12 mb-4"
                  >
                    <h1>
                      Jelentkezz önkéntesnek
                    </h1>
                  </div>
                </div>
                <div
                  class="row mb-5"
                >
                  <div
                    class="col-12 col-lg-6 align-self-center mb-4 mb-lg-0"
                  >
                    <p>
                      <strong>
                        Cím: 
                      </strong>
                    </p>
                    <p>
                      Beogradska 3-5
                      <br />
                      24420 Magyarkanizsa
                      <br />
                      Szerbia
                    </p>
                    <p
                      class="mt-3"
                    >
                      <strong>
                        Telefon: 
                      </strong>
                    </p>
                    <p>
                      +381 00 000 000
                    </p>
                    <p
                      class="mt-3"
                    >
                      <strong>
                        Email: 
                      </strong>
                    </p>
                    <p>
                      mancsmagyarkanizsa@gmail.com
                    </p>
                    <p
                      class="mt-3"
                    >
                      <strong>
                        Facebook: 
                      </strong>
                    </p>
                    <p>
                      <a
                        href="/"
                      >
                        Mancs Állatvédő Egyesület
                      </a>
                    </p>
                  </div>
                  <div
                    class="col-12 col-lg-6"
                  >
                    <form
                      class="contact-form p-4"
                    >
                      <h4
                        class="mb-3"
                      >
                        Adataid:
                      </h4>
                      <div
                        class="row"
                      >
                        <div
                          class="col-12"
                        >
                          <div
                            class="form-control-container"
                          >
                            <div
                              class="form-icon-container"
                            >
                              <i
                                class="fa-solid fa-user"
                              />
                              <div
                                class="tooltip-container"
                              >
                                <p>
                                  Vezetéknév
                                  <br />
                                   Mező kitöltése kötelező!
                                </p>
                              </div>
                            </div>
                            <input
                              class="form-control"
                              id="contactLastName"
                              name="c_last_name"
                              placeholder="Vezetéknév"
                              required=""
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                        <div
                          class="col-12"
                        >
                          <div
                            class="form-control-container"
                          >
                            <div
                              class="form-icon-container"
                            >
                              <i
                                class="fa-solid fa-user"
                              />
                              <div
                                class="tooltip-container"
                              >
                                <p>
                                  Keresztnév
                                  <br />
                                   Mező kitöltése kötelező!
                                </p>
                              </div>
                            </div>
                            <input
                              class="form-control"
                              id="contactFirstName"
                              name="c_first_name"
                              placeholder="Keresztnév"
                              required=""
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                        <div
                          class="col-12"
                        >
                          <div
                            class="form-control-container"
                          >
                            <div
                              class="form-icon-container"
                            >
                              <i
                                class="fa-solid fa-at"
                              />
                              <div
                                class="tooltip-container"
                              >
                                <p>
                                  Email
                                  <br />
                                   Mező kitöltése kötelező!
                                </p>
                              </div>
                            </div>
                            <input
                              class="form-control"
                              id="contactEmail"
                              name="c_email"
                              placeholder="Email"
                              required=""
                              type="email"
                              value=""
                            />
                          </div>
                        </div>
                        <div
                          class="col-12"
                        >
                          <div
                            class="form-control-container"
                          >
                            <div
                              class="form-icon-container"
                            >
                              <i
                                class="fa-solid fa-phone"
                              />
                              <div
                                class="tooltip-container"
                              >
                                <p>
                                  Telefonszám
                                  <br />
                                   Mező kitöltése kötelező!
                                </p>
                              </div>
                            </div>
                            <input
                              class="form-control"
                              id="contactPhone"
                              name="c_phone"
                              placeholder="Telefonszám"
                              required=""
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                        <div
                          class="col-12"
                        >
                          <div
                            class="form-control-container"
                          >
                            <div
                              class="form-icon-container"
                            >
                              <i
                                class="fa-solid fa-message"
                              />
                              <div
                                class="tooltip-container"
                              >
                                <p>
                                  Üzenet
                                  <br />
                                   Mező kitöltése kötelező!
                                </p>
                              </div>
                            </div>
                            <textarea
                              class="form-control"
                              id="contactMessage"
                              name="c_message"
                              placeholder="Üzenet"
                              required=""
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class="row justify-content-end"
                      >
                        <div
                          class="col-6 mt-3"
                        >
                          <button
                            class="button call-to-action-button w-100"
                          >
                            Küldés
                          </button>
                        </div>
                      </div>
                    </form>
                  </div>
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col-12"
                  >
                    <h4
                      class="mb-4"
                    >
                      Támogass minked önkéntes adományoddal:
                    </h4>
                    <p>
                      <strong>
                        Számla tulajdonos: 
                      </strong>
                      Mancs Állatvédő Egyesület
                    </p>
                    <p
                      class="mt-1"
                    >
                      <strong>
                        Bank neve: 
                      </strong>
                      OTP Bank
                    </p>
                    <p
                      class="mt-1"
                    >
                      <strong>
                        Belföldi számláról utalás: 
                      </strong>
                      00000000 - 00000000 - 00000000
                    </p>
                    <p
                      class="mt-1"
                    >
                      <strong>
                        Külfőldi számláról utalás: 
                      </strong>
                      SRB62 00000000 - 00000000 - 00000000
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="container-fluid footer-container"
  >
    <div
      class="row"
    >
      <div
        class="col-12"
      >
        <div
          class="container py-5"
        >
          <div
            class="row"
          >
            <div
              class="col-12 col-md-5 col-lg-4 mb-4 mb-md-0"
            >
              <h5>
                Mancs állatvédő egyesület
              </h5>
              <p
                class="footer-description"
              >
                Tevékenységeink és céljaink magába foglalják az egyhagyott, kidobott, megtalált kölyök és felnőtt állatok mentését, gondozását, nevelését és legfőképp gazdához jutását. Hisszuk, hogy minden állatnak van megfelelő gazdája, mi segítünk megtalálni!
              </p>
            </div>
            <div
              class="col-6 col-md-4 offset-lg-1 useful-links"
            >
              <h5>
                Hasznos linkek
              </h5>
              <a
                href="/"
              >
                Főoldal
              </a>
              <a
                href="/lakok"
              >
                Lakóink
              </a>
              <a
                href="/hirek"
              >
                Hírek
              </a>
              <a
                href="/csatlakozz"
              >
                Csatlakozz!
              </a>
              <a
                href="/kapcsolat"
              >
                Kapcsolat
              </a>
            </div>
            <div
              class="col-6 col-md-3 social-links-container"
            >
              <h5>
                Kapcsolat
              </h5>
              <p>
                Beogradska 3-5
                <br />
                24420 Magyarkanizsa
                <br />
                Szerbia
              </p>
              <h5
                class="mt-4"
              >
                Kövess minket
              </h5>
              <a
                href="/facebook"
                style="text-decoration: none;"
              >
                <i
                  class="fa-brands fa-facebook"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="container-fluid copyright-strip-container"
  >
    <div
      class="row"
    >
      <div
        class="col-12"
      >
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col-12 text-center"
            >
              <p>
                © 2023. Minden jog fenntartva. Mancs Állatvédő Egyesület Magyarkanizsa
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
